packages/netdog/netdog.spec

```diff
@@ -0,0 +1,133 @@
+%global _cross_first_party 1
+%undefine _debugsource_packages
+
+Name: %{_cross_os}netdog
+Version: 0.1.0
+Release: 0%{?dist}
+Summary: Bottlerocket network configuration helper
+License: Apache-2.0 OR MIT
+URL: https://github.com/bottlerocket-os/bottlerocket
+
+Source0: netdog-tmpfiles.conf
+
+Source10: run-netdog.mount
+Source11: write-network-status.service
+Source12: generate-network-config.service
+Source13: disable-udp-offload.service
+
+Source20: 00-resolved.conf
+
+BuildRequires: %{_cross_os}glibc-devel
+Requires: %{_cross_os}netdog
+Requires: (%{_cross_os}netdog-systemd-networkd if %{_cross_os}image-feature(systemd-networkd))
+Requires: (%{_cross_os}netdog-wicked if %{_cross_os}image-feature(no-systemd-networkd))
+
+%description
+%{summary}.
+
+%package -n %{_cross_os}netdog-common
+Summary: Common configuration for Bottlerocket's network configuration helper
+Requires: %{_cross_os}hostname-reverse-dns
+Requires: (%{_cross_os}hostname-imds if %{_cross_os}variant-platform(aws))
+%description -n %{_cross_os}netdog-common
+%{summary}.
+
+%package -n %{_cross_os}netdog-systemd-networkd
+Summary: Bottlerocket network configuration helper
+Provides: %{_cross_os}netdog = 2
+Requires: %{_cross_os}netdog-common
+Requires: %{_cross_os}systemd-networkd
+Requires: %{_cross_os}systemd-resolved
+Requires: %{_cross_os}netdog
+Conflicts: (%{_cross_os}netdog-wicked or %{_cross_os}image-feature(no-systemd-networkd))
+%description -n %{_cross_os}netdog-systemd-networkd
+%{summary}.
+
+%package -n %{_cross_os}netdog-wicked
+Summary: Bottlerocket network configuration helper
+Provides: %{_cross_os}netdog = 1
+Requires: %{_cross_os}netdog-common
+Requires: %{_cross_os}wicked
+Requires: %{_cross_os}netdog
+Conflicts: (%{_cross_os}netdog-systemd-networkd or %{_cross_os}image-feature(systemd-networkd))
+%description -n %{_cross_os}netdog-wicked
+%{summary}.
+
+%package -n %{_cross_os}hostname-reverse-dns
+Summary: Reverse DNS Hostname detector
+%description -n %{_cross_os}hostname-reverse-dns
+%{summary}
+
+%package -n %{_cross_os}hostname-imds
+Summary: IMDS Hostname detector
+%description -n %{_cross_os}hostname-imds
+%{summary}
+
+%prep
+%setup -T -c
+%cargo_prep
+
+%build
+mkdir bin
+
+echo "** Build Dogtag Hostname Helpers"
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p dogtag \
+    --bins \
+    --target-dir=${HOME}/.cache/dogtag
+
+echo "** Build Netdog Binaries"
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p netdog \
+    --features default \
+    --target-dir=${HOME}/.cache/networkd
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p netdog \
+    --features wicked \
+    --target-dir=${HOME}/.cache/wicked
+
+%install
+install -d %{buildroot}%{_cross_libexecdir}/hostname-resolvers
```

Comment:
I would pick a consistent phrase - either "hostname resolvers" or "hostname detectors" ("detectors" is in the package description and would be easiest to change).